tests/web/js_utils.cc

```diff
@@ -13,198 +14,311 @@
+          emp.PassNestedVectorToJS(); // {{1,2,3},{4,5,6}}}
+          const incoming = emp_i.__incoming_array;
+          chai.assert.equal(incoming[0][0][0], 1);
+          chai.assert.equal(incoming[0][0][1], 2);
+          chai.assert.equal(incoming[0][0][2], 3);
+          chai.assert.equal(incoming[0][1][0], 4);
+          chai.assert.equal(incoming[0][1][1], 5);
+          chai.assert.equal(incoming[0][1][2], 6);
+        });
+        it("should pass emp::vector<int> to javascript", function() {
+          emp.PassIntVectorToJS(); // {5,1,2,3,6}
+          const incoming = emp_i.__incoming_array;
+          chai.assert.deepEqual(incoming, [5, 1, 2, 3, 6]);
+        });
+        it("should pass emp::vector<std::string> to javascript", function() {
+          emp.PassStringVectorToJS(); // {"a", "vector", "of", "strings"}
+          const incoming = emp_i.__incoming_array;
+          chai.assert.deepEqual(incoming, ['a', 'vector', 'of', 'strings']);
+        });
+        it("should pass emp::array<int, N> to javascript", function() {
+          emp.PassArrayIntToJS(); // {{10,30,60}}
+          const incoming = emp_i.__incoming_array;
+          chai.assert.deepEqual(incoming, [10, 30, 60]);
+        });
+        it("should pass emp::array<JSDataObject, N> to javascript", function() {
+          emp.PassArrayJSDataObjectToJS();
+          const incoming = emp_i.__incoming_array;
+          chai.assert.deepEqual(incoming, [{val: 10, word: 'hi', val2: 4.4}, {val: 40, word: 'hi2', val2: 11.2}]);
+        });
+        it("should pass emp::array<emp::array<std::string, N0>, N1> to javascript", function() {
+          emp.PassNestedArrayStringToJS();
+          const incoming = emp_i.__incoming_array;
+          // chai.assert.deepEqual(incoming, ['do', 'strings', 'work', 'in', 'arrays?']);
+          chai.assert.equal(incoming[0][3], "in")
+        });
+        it("should pass emp::array<emp::array<emp::array<int, N0>, N1>, N2> to javascript", function() {
+          emp.PassNestedArrayIntToJS(); // {{{{{{0,0}}}}, {{{{0,10}}}}, {{{{10,10}}}}, {{{{20,20}}}}, {{{{30, 30}}}}}}
+          const incoming = emp_i.__incoming_array;
+          chai.assert.equal(incoming[4][0][0], 30);
+        });
+        it("should pass emp::array<emp::array<JSDataObject, N0>, N1> to javascript", function() {
+          emp.PassNestedArrayJSDataObjectToJS(); // {{{{test_obj_1, test_obj_2}}, {{test_obj_2, test_obj_2}}}};
+          const incoming = emp_i.__incoming_array;
+          chai.assert.equal(incoming[1][0].val, 40);
+          chai.assert.equal(incoming[1][0].val2, 11.2);
+        });
+      });
+    });
+  }
+
 };
 
-int main(int argc, char* argv[]) {
-  //emp::vector<std::string> args = emp::cl::args_to_strings(argc, argv);
-  //const bool verbose = emp::cl::use_arg(args, "-v");
-  [[maybe_unused]] bool verbose = true;
 
+// Because we want to check values on the C++ end, this test relies on emp_assert to check values.
```

Comment:
I /think/ this comment is out of date? Looks like asserts rather than emp_asserts